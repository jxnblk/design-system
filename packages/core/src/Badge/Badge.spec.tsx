import { Badge } from '..'

describe('Badge', () => {
  let consoleError: typeof console.error
  beforeEach(() => {
    consoleError = console.error
    console.error = jest.fn()
  })
  afterEach(() => (console.error = consoleError))

  test('renders', () => {
    const json = rendererCreateWithTheme(<Badge />).toJSON()
    expect(json).toMatchSnapshot()
  })
})
