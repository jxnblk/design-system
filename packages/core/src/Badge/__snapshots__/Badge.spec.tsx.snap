// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge bg blue sets background-color and color 1`] = `
.c0 {
  display: inline-block;
  font-size: 12px;
  font-weight: 700;
  line-height: 1.25;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  text-transform: uppercase;
  line-height: 1.4;
  background-color: #0068ef;
  color: #fff;
  background-color: #0068ef;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  border-radius: 9999px;
}

<div
  className="c0"
  size="medium"
/>
`;

exports[`Badge bg green sets background-color and color 1`] = `
.c0 {
  display: inline-block;
  font-size: 12px;
  font-weight: 700;
  line-height: 1.25;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  text-transform: uppercase;
  line-height: 1.4;
  background-color: #0a0;
  color: #fff;
  background-color: #0a0;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  border-radius: 9999px;
}

<div
  className="c0"
  size="medium"
/>
`;

exports[`Badge bg lightBlue sets background-color and color 1`] = `
.c0 {
  display: inline-block;
  font-size: 12px;
  font-weight: 700;
  line-height: 1.25;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  text-transform: uppercase;
  line-height: 1.4;
  background-color: #e8f2ff;
  color: #049;
  background-color: #e8f2ff;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  border-radius: 9999px;
}

<div
  className="c0"
  size="medium"
/>
`;

exports[`Badge bg lightGreen sets background-color and color 1`] = `
.c0 {
  display: inline-block;
  font-size: 12px;
  font-weight: 700;
  line-height: 1.25;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  text-transform: uppercase;
  line-height: 1.4;
  background-color: #ecf7ec;
  color: #060;
  background-color: #ecf7ec;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  border-radius: 9999px;
}

<div
  className="c0"
  size="medium"
/>
`;

exports[`Badge bg lightRed sets background-color and color 1`] = `
.c0 {
  display: inline-block;
  font-size: 12px;
  font-weight: 700;
  line-height: 1.25;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  text-transform: uppercase;
  line-height: 1.4;
  background-color: #fbebeb;
  color: #800;
  background-color: #fbebeb;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  border-radius: 9999px;
}

<div
  className="c0"
  size="medium"
/>
`;

exports[`Badge bg orange sets background-color and color 1`] = `
.c0 {
  display: inline-block;
  font-size: 12px;
  font-weight: 700;
  line-height: 1.25;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  text-transform: uppercase;
  line-height: 1.4;
  background-color: #f68013;
  color: #001833;
  background-color: #f68013;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  border-radius: 9999px;
}

<div
  className="c0"
  size="medium"
/>
`;

exports[`Badge bg red sets background-color and color 1`] = `
.c0 {
  display: inline-block;
  font-size: 12px;
  font-weight: 700;
  line-height: 1.25;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  text-transform: uppercase;
  line-height: 1.4;
  background-color: #c00;
  color: #fff;
  background-color: #c00;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  border-radius: 9999px;
}

<div
  className="c0"
  size="medium"
/>
`;

exports[`Badge can escape preset: bg lightBlue sets background-color and color text sets color 1`] = `
.c0 {
  display: inline-block;
  font-size: 12px;
  font-weight: 700;
  line-height: 1.25;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  text-transform: uppercase;
  line-height: 1.4;
  background-color: #e8f2ff;
  color: #001833;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  border-radius: 9999px;
}

<div
  className="c0"
  color="text"
  size="medium"
/>
`;

exports[`Badge non-preset: bg text sets background-color and color white sets color 1`] = `
.c0 {
  display: inline-block;
  font-size: 12px;
  font-weight: 700;
  line-height: 1.25;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  text-transform: uppercase;
  line-height: 1.4;
  background-color: #001833;
  color: #fff;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  border-radius: 9999px;
}

<div
  className="c0"
  color="white"
  size="medium"
/>
`;

exports[`Badge renders 1`] = `
.c0 {
  display: inline-block;
  font-size: 12px;
  font-weight: 700;
  line-height: 1.25;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  text-transform: uppercase;
  line-height: 1.4;
  background-color: #f4f6f8;
  color: #001833;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  border-radius: 9999px;
}

<div
  className="c0"
  size="medium"
/>
`;

exports[`Badge renders small 1`] = `
.c0 {
  display: inline-block;
  font-size: 10px;
  font-weight: 700;
  line-height: 1.4;
  line-height: 1.25;
  background-color: #f4f6f8;
  color: #001833;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  border-radius: 9999px;
}

<div
  className="c0"
  size="small"
/>
`;
